### PR TITLE
refactor(ts): enable noImplicitAny on mutation.ts

### LINKED ISSUE #427: Convert library to TypeScript
This needs to happen in steps:
- [x] Rename all the files to `*.ts` and introduce a compile step
- [x] Get the linter and formatter working
- [ ] Enable noImplicitAny and types for all src/ files
- [ ] Update the package.json to ship the types
- [ ] Enable noImplicitAny everywhere

### PR DESCRIPTION
**Pull request by joecervino**
Fixes #427 (2/21ish files)

Notes:
1. I branched this off of the `run-fix` branch to make sure I'm linting properly. Hope that's cool.
2. This was a tricky one, no online docs and little in-repo docs, so looking forward to learning :)

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by joecervino** [comment]
Boo docs, boo 👎

File: src/mutation.ts [hunk 1]
```diff
@@ -14,10 +14,49 @@
  * limitations under the License.
  */
 
+import {Metadata} from '@google-cloud/common';
```
Comment (callmehiphop) [inline src/mutation.ts]: Pretty sure we should never use this type in grpc clients.

File: src/mutation.ts [hunk 2]
```diff
@@ -242,7 +285,8 @@ export class Mutation {
    * ]);
    * @private
    */
-  static encodeDelete(data?) {
+  static encodeDelete(data?: Metadata): EncodeDeleteDataResponse
```
Comment (callmehiphop) [inline src/mutation.ts]: Usually `encode` methods are just formatting request objects. I think you could use [`IDeleteFromColumn`](https://github.com/googleapis/nodejs-bigtable/blob/master/proto/bigtable.d.ts#L3336) here, but moving forward I think we should avoid using `Response` for these kinds of methods.

File: src/mutation.ts [hunk 3]
```diff
@@ -154,8 +196,9 @@ export class Mutation {
    * // ]
    * @private
    */
-  static encodeSetCell(data) {
-    const mutations: any[] = [];
+  static encodeSetCell(data: Metadata): EncodeSetCellResponse[] {
```
Comment (callmehiphop) [inline src/mutation.ts]: Can we use [`ISetCell`](https://github.com/googleapis/nodejs-bigtable/blob/master/proto/bigtable.d.ts#L3228) here?

File: src/mutation.ts [hunk 4]
```diff
@@ -292,7 +336,7 @@ export class Mutation {
    * @returns {object}
    * @private
    */
-  static parse(mutation) {
+  static parse(mutation: Mutation): JsonObj {
```
Comment (callmehiphop) [inline src/mutation.ts]: Should this be [`IMutateRowRequest`](https://github.com/googleapis/nodejs-bigtable/blob/master/proto/bigtable.d.ts#L709) ?

File: src/mutation.ts [hunk 5]
```diff
@@ -314,7 +358,7 @@ export class Mutation {
    * // }
    * @private
    */
-  static parseColumnName(column) {
+  static parseColumnName(column: string): ParseColumnNameResponse {
```
Comment (callmehiphop) [inline src/mutation.ts]: Can we rename this to something like `Column`?

File: src/mutation.ts [hunk 6]
```diff
@@ -329,7 +373,8 @@ export class Mutation {
    * @returns {object}
    * @private
    */
-  toProto() {
+  toProto(): JsonObj {
+    // tslint:disable-next-line no-any
```
Comment (callmehiphop) [inline src/mutation.ts]: I think this should be [`IMutateRowRequest`](https://github.com/googleapis/nodejs-bigtable/blob/master/proto/bigtable.d.ts#L709), should be able to drop the `no-any` comment too.

File: src/mutation.ts (hunk 2 above)
Comment (joecervino) [reply src/mutation.ts]: Just curious, want to learn this right - is there something specific about `static` methods that mean I should avoid a `Response` return annotation or is it something more general about `Response`s?

Comment (callmehiphop) [reply src/mutation.ts]: More general for `Response`, IMO we should only use that word to describe actual response payloads, where this method I think is just something we use internally to transform an object into the expected proto format.

Comment (joecervino) [reply src/mutation.ts]: Makes sense, cool. Noted 👍

**Comment by JustinBeckwith** [comment]
@callmehiphop can this get another look?

**Pull request merged**